drone/build_info.go

```diff
@@ -32,16 +32,28 @@ func buildInfo(c *cli.Context) error {
 	if err != nil {
 		return err
 	}
-	number, err := strconv.Atoi(c.Args().Get(1))
-	if err != nil {
-		return err
-	}
+	buildArg := c.Args().Get(1)
 
 	client, err := newClient(c)
 	if err != nil {
 		return err
 	}
 
+	var number int
+	if buildArg == "last" {
+		// Fetch the build number from the last build
+		build, err := client.BuildLast(owner, name, "")
+		if err != nil {
+			return err
+		}
+		number = build.Number
+	} else {
+		number, err = strconv.Atoi(buildArg)
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
also for consistency [1], we should let the user specify the branch using the `--branch` flag. This can default to an empty string, but if provided, can be passed to the `client.BuildLast` function

```diff
-build, err := client.BuildLast(owner, name, "")
+build, err := client.BuildLast(owner, name, c.String("branch"))
```

[1] https://github.com/drone/drone/blob/master/drone/build_last.go#L45